src/main/java/racingcar/game/RacingGame.java

```diff
@@ -0,0 +1,75 @@
+package racingcar.game;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import racingcar.domain.Car;
+
+import static racingcar.util.ValidationUtil.*;
+import static racingcar.util.RandomUtil.*;
+
+public class RacingGame {
+
+    private Set<Car> cars = new HashSet<>();
+    private int totalAttempt;
+
+    public void initCarNames(String carNames) {
+        String[] names = carNames.split(",");
+        isDuplicated(names);
+
+        for (String name : names) {
+            cars.add(new Car(name));
+        }
+    }
+
+    public void initTotalAttempt(String attempt) {
+        validateAttempt(attempt);
+        totalAttempt = Integer.parseInt(attempt);
+    }
+
+    public String play() {
+        StringBuilder totalExecutionResult = new StringBuilder();
+
+        while (totalAttempt-- > 0) {
+            run();
+            totalExecutionResult.append(generateExecutionResult() + "\n");
+        }
+
+        return totalExecutionResult.toString();
+    }
+
+    public List<String> selectWinners() {
+        int maxPosition = getMaxPosition();
+
+        return cars.stream()
+                .filter(c -> c.isWinner(maxPosition))
+                .map(Car::getName)
+                .collect(Collectors.toList());
+    }
+
+    private void run() {
+        for (Car car : cars) {
+            car.moveOrNot(generateRandomNumber());
+        }
+    }
+
+    private String generateExecutionResult() {
+        StringBuilder statement = new StringBuilder();
+
+        for (Car car : cars) {
+            statement.append(car.toString());
+        }
+
+        return statement.toString();
+    }
+
+    private int getMaxPosition() {
+        return cars.stream()
+                .sorted(Comparator.comparing(Car::getPosition).reversed())
+                .findFirst()
+                .get()
+                .getPosition();
+
+    }
```

Comment:
stream 사용 👍

그런데 아래 처럼 구현하면 더욱 가독성이 좋지 않을까요?

자바 stream api를 조금 더 학습해보시면 좋을 것 같아요.
자연스럽게 읽히는 코드를 더욱 쉽게 작성할 수 있어요!

```suggestion
        return cars.stream()
                .mapToInt(Car::getPosition)
                .max()
                .orElse(0);
    }
```

Reply:
max를 이용하니 정렬할 필요가 없군요.. mapToInt로 position을 먼저 추출한 뒤 최댓값을 구했다면 getPosition을 한번 더 호출할 필요도 없구요..!  부끄럽습니다ㅜㅜ 더 공부하겠습니다.